src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/MethodToStateMachineRewriter.cs

```diff
@@ -621,6 +622,52 @@ public override BoundNode VisitBlock(BoundBlock node)
             return PossibleIteratorScope(node.Locals, () => (BoundStatement)base.VisitBlock(node));
         }
 
+        public override BoundNode VisitScope(BoundScope node)
```

Comment:
Lambda frames are build for life-time scopes, the added node is a visibility scope, no frames are build for it as that would not allow us to expand the lifetime of the locals that belong to it.